### ISSUE TITLE
Semver doesn't seem to handle multiple-workspace projects well

### ISSUE THREAD

**Issue by orium** [rust-lang-nursery/rust-semverver#67]
For instance if you run `cargo semver -S url:1.0.0 -C url:1.7.1` this will happen:

```text
thread 'main' panicked at 'assertion failed: self.in_new_crate(new)', src/semcheck/mapping.rs:122:9
```

The item that cause this lives in the `percent_encoding` workspace.

**Comment by ibabushkin** [comment]
Yes, that is indeed an issue. I should have more time to work on the tool soon though :)

**Comment by ibabushkin** [comment]
Okay, so funnily enough we handle such cases somewhat gracefully (as in do not crash and do reasonable(?) computations) when both the old and the new crate are on-disk (as in, not specified via `-C` or `-S`). Other cases fail, because I presume the other crates in the workspace that are often dependencies get dragged in somehow (and possibly translated and so on).
